### DISCUSSION TITLE
Built one!

### DISCUSSION THREAD

**Discussion by RussellSenior** [picocomputer/community#135] [Q&A]
I got a set of parts from Mouser on Friday, picked up the board and case yesterday afternoon at about 2::30 pm and put it together last night:
https://photos.app.goo.gl/gVx6JYac1GBZWsRx6.

First impressions: It works! Second impression, the USB drive seems very slow. PHI2 is set to 8000 kHz, RESB: 0.001 ms (auto), but ]LOAD tetricks.rp6502 from a 8GB HP v165w (rev 1100) USB drive takes about 90 seconds. LOAD star_swarms.rp6502 takes 3 minutes. Is that normal? I tried a 4GB USB drive before and it was also slow.RIA Version 0.9 and VGA Version 0.9.

**Comment by RussellSenior** [comment]
![PXL_20241201_090034754 MP](https://github.com/user-attachments/assets/50f11165-ae74-4e5a-9321-034b7d2c4454)
![PXL_20241201_090147779](https://github.com/user-attachments/assets/787409f2-5ed9-44f6-b888-1700ee2a04ba)
![PXL_20241201_104522778](https://github.com/user-attachments/assets/87c70499-5509-4765-8387-65a73bcd18b0)

**Comment by rumbledethumps** [comment]
Should only take a second or three to load those files. Try a different hub if it's not the drive.

**Reply by RussellSenior** [reply]
fwiw, this is the hub involved: https://www.adafruit.com/product/2991

**Reply by RussellSenior** [reply]
I tried two other hubs, connected to a passive usb otg cable, and no difference. One question that didn't seem to be addressed in the Episode 10 assembly video and that was the three pin sockets under the RIA pico. I populated the three debug pins and those are connected. The other three holes didn't seem to align with anything on the pico, and they seem to parallel the other three, so I left those holes unpopulated. I'm wondering if the fact that the RIA is connected to the debug pins might be relevant to my slowness?

**Reply by rumbledethumps** [reply]
OTG USB with TinyUSB isn't as robust as a modern desktop OS. Have you swapped the keyboard and mouse too?

**Reply by RussellSenior** [reply]
I've been using a Logitech K400+ wireless keyboard/trackpad thingie. I tried a different keyboard (wired Gateway KB-2961) has the same USB storage access slowness.

**Reply by RussellSenior** [reply]
I tested my debug pin theory by programming another pi pico as a RIA without the debug pins populated (so, no connection). Super slowness persists. USB attached keyboard works okay, but "load basic.rp6502" (11926 bytes) takes 68 seconds, about 175 bytes/second. I have tried four different hubs, and two different keyboards, with no change to the load speed.

**Reply by rumbledethumps** [reply]
Check if the `install` command is also slow.

**Reply by RussellSenior** [reply]
`install basic.rp6502` takes about the same time as `load basic.rp6502`. Running `basic` rom from the RIA is fast, starts in less than a second.

**Reply by rumbledethumps** [reply]
It all points to a problem with the USB bus. You need to consider every device, cable, power supply, and interfering source. Nothing like this has been reported before.

**Reply by RussellSenior** [reply]
Someone reported a particular OTG hub worked for them, I have a couple on order. If I take the hub out of the loop entirely, and plug a USB thumb directly into the OTG cable (converting to a single USB A host port), I see the same slowness. I have the power coming from my laptop USB port now (it has been on a rated-1.2A 5V power supply) so I can see the serial console.

```
status
Picocomputer 6502
RIA Version 0.9
VGA Version 0.9
USB: 1 device
1: MSC 7.7 GB hp       v165w            rev 1100
```

It can't be the hub's fault, because I see same thing with no hub. And, when plugged in, the usb keyboard seems to work fine.

**Reply by RussellSenior** [reply]
Aha! Tried a third USB storage device (actually an 4GB SD card in an adapter) and suddenly everything is reasonable fast! Super weird that it didn't like my name-brand USB2 thumb drive.

**Comment by rumbledethumps** [comment]
I got the USB drive from Russel today. It worked on the Picocomputer but took about 10 seconds to mount. I checked it under Windows and it also took 10 seconds to mount. It was partitioned for EFI, so I tried to format it under windows, which failed. Tried diskpart, failed. Finally I changed it to GPT, formatted, changed back to MBR, formatted again, and it worked. Still takes 10 seconds to mount on either Windows or Picocomputer.

Conclusion: This drive is cursed.

**Reply by RussellSenior** [reply]
I think it works reasonably well on Linux. At least I don't recall weirdness there.

**Comment by RussellSenior** [comment]
For example, a different but also discovered to be problematic usb drive: Bus 003 Device 012: ID 03f0:3d07 HP, Inc v115w:

If I insert it on Ubuntu 24.04, it automounts in about 4 seconds (counted outloud). If I umount it and then run:

> $ time sudo mount /dev/sda1 /mnt
> 
> real	0m0.067s
> user	0m0.004s
> sys	0m0.005s
> 
> $ mount | grep sda1
> /dev/sda1 on /mnt type ext4 (rw,relatime)
> 
> $ umount /dev/sda1
> $ sudo mkfs.vfat /dev/sda1
> mkfs.fat 4.2 (2021-01-31)
> $ time sudo mount /dev/sda1 /mnt
> 
> real	0m0.070s
> user	0m0.005s
> sys	0m0.008s
>
